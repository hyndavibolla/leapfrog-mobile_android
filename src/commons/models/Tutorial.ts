import { ReactNode } from 'react';
import { StyleProp, ViewStyle } from 'react-native';

export interface Tutorial {
  isParent?: boolean;
  isHidden?: boolean;
  parentId?: number;
  scrollToDown?: number;
}

export interface TutorialElementPosition extends Tutorial {
  x: number;
  y: number;
  children: ReactNode;
  style: StyleProp<ViewStyle>;
}
