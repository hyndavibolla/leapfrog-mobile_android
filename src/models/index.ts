import * as GeneralModel from './general';
import * as UserModel from './user';
import * as ActivityModel from './activity';
import * as OfferModel from './offer';
import * as GameModel from './game';
import * as MissionModel from './mission';
import * as RewardModel from './reward';
import * as StreakModel from './streak';
import * as PromptModel from './prompt';
import * as CardLink from './cardLink';

export { GeneralModel, UserModel, ActivityModel, OfferModel, GameModel, MissionModel, RewardModel, StreakModel, PromptModel, CardLink };
