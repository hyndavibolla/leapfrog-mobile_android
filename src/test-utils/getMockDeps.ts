import { ENV } from '_constants';
import { GeneralModel, MissionModel } from '_models';
import { RUNTIME_ENV } from '_models/general';
import { ICrashlyticsService } from '_services/CrashlyticsService';
import { IHttpService } from '_services/HttpService';
import { IPerformanceTrackingService } from '_services/PerformanceTrackingService';
import { getInitialState } from '_state_mgmt/GlobalState';
import {
  getUser,
  getGame,
  getActivity_1,
  getActivity_2,
  getActivity_3,
  getMissionCategory_1,
  getMissionCategory_2,
  getMissionCategory_3,
  getMissionKeyword_1,
  getMissionKeyword_2,
  getMissionKeyword_3,
  getMission_1,
  getRewardConfig_1,
  getSlideObject_1,
  getSlideObject_2,
  getSlideCategoryIdList,
  getStreak_1,
  getQuestion_1,
  getQuestion_2,
  getQuestion_3,
  getAnswer_1,
  getAnswer_2,
  getAnswer_3,
  getNewOnMaxMessage_1,
  getNewOnMaxMessage_2,
  getNewOnMaxMessage_3,
  getSailthruMessage_1,
  getLinkedCards_1,
  getLinkedCards_2,
  getLinkedCards_3,
  getLinkedCards_4,
  getNewOnMaxMessage_4,
  getGiftCard,
  getGiftCardDetail,
  getGiftCardBalance
} from '_test_utils/entities';
import { createUUID } from '_utils/create-uuid';

export const getMockDeps = (initialStateGetter = getInitialState): GeneralModel.Deps => ({
  apiService: {
    isTokenInvalid: false,
    getDeviceIdAsync: async () => createUUID(),
    isTokenSetCurrentlyReady: () => true,
    setTokenSet: jest.fn(),
    authenticate: jest
      .fn()
      .mockResolvedValue({ accessToken: 'accessToken', refreshToken: 'refreshToken', accessTokenExpiryTime: 'accessTokenExpiryTime', deviceId: 'deviceId' }),
    enrollCard: jest.fn().mockResolvedValue(undefined),
    fetchRefId: jest.fn().mockResolvedValue('test-ref-id'),
    fetchUserProfile: jest.fn().mockResolvedValue(getUser()),
    fetchGameState: jest.fn().mockResolvedValue(getGame()),
    fetchRewardConfig: jest.fn().mockResolvedValue(getRewardConfig_1()),
    fetchActivityHistory: jest.fn().mockResolvedValue([getActivity_1(), getActivity_2(), getActivity_3()]),
    fetchGiftCardList: jest.fn().mockResolvedValue([getGiftCard()]),
    fetchGiftCardDetail: jest.fn().mockResolvedValue(getGiftCardDetail()),
    fetchGiftCardBalance: jest.fn().mockResolvedValue(getGiftCardBalance()),
    fetchWalletStats: jest.fn().mockResolvedValue({
      totalLifetimeEarnedPoints: 15550,
      monthlySummaries: {
        '2020-02': 1500,
        '2020-03': 2000,
        '2020-04': 100,
        '2020-05': 950,
        '2020-06': 1000,
        '2020-07': 1500
      }
    }),
    fetchMissionCategoryList: jest.fn().mockResolvedValue([getMissionCategory_1(), getMissionCategory_2(), getMissionCategory_3()]),
    fetchMissionKeywordList: jest.fn().mockResolvedValue({
      [MissionModel.KeywordType.CATEGORY]: [getMissionKeyword_1().value],
      [MissionModel.KeywordType.BRAND]: [getMissionKeyword_2().value, getMissionKeyword_3().value]
    }),
    fetchMissionList: jest.fn(async ({ listType }) => ({ userId: 'buttonId', missions: [getMission_1()], title: 'mission endpoint title', listType })),
    registerDeviceInfo: jest.fn(),
    registerUserActivity: jest.fn(),
    logout: jest.fn(),
    getSYWOnlineToken: jest.fn().mockResolvedValue('test-token'),
    sendValidationEmail: jest.fn().mockResolvedValue(undefined),
    fetchStreakList: jest.fn().mockResolvedValue([getStreak_1()]),
    fetchPromptList: jest.fn().mockResolvedValue({
      promptQuestions: [getQuestion_1(), getQuestion_2(), getQuestion_3()],
      answeredQuestions: [getAnswer_1(), getAnswer_2(), getAnswer_3()]
    }),
    updatePromptList: jest.fn(),
    unregisterDeviceInfo: jest.fn(),
    fetchLinkedCardsList: jest.fn().mockResolvedValue({ linkedCards: [getLinkedCards_1(), getLinkedCards_2(), getLinkedCards_3(), getLinkedCards_4()] }),
    activateLocalOffer: jest.fn(),
    fetchAppleWalletPass: jest.fn().mockResolvedValue('test-pass'),
    updateCardStatus: jest.fn()
  } as any,
  appService: {
    getDeviceIdAsync: async () => createUUID()
  } as any,
  crashlyticsService: {
    initialize: jest.fn(),
    log: jest.fn(),
    recordError: jest.fn(),
    setCrashlyticsCollectionEnabled: jest.fn()
  } as ICrashlyticsService,
  httpService: {
    fetch: jest.fn()
  } as IHttpService,
  performanceTrackingService: {
    initialize: jest.fn(),
    measuredFetch: jest.fn(),
    setPerformanceCollectionEnabled: jest.fn()
  } as IPerformanceTrackingService,
  raiseService: {
    fetchSlideBrand: jest.fn().mockResolvedValue([getSlideObject_1(), getSlideObject_2()]),
    fetchSlideCategoryIdList: jest.fn().mockResolvedValue(getSlideCategoryIdList())
  } as any,
  nativeHelperService: {
    alert: {
      alert: jest.fn(),
      prompt: jest.fn()
    },
    appState: { addEventListener: jest.fn(), currentState: 'active' },
    storage: {
      get: jest.fn().mockResolvedValue(null),
      set: jest.fn().mockResolvedValue(null),
      remove: jest.fn().mockResolvedValue(null)
    },
    clipboard: {
      getString: jest.fn(),
      setString: jest.fn()
    },
    dimensions: {
      getWindowWidth: jest.fn().mockReturnValue(750),
      getWindowHeight: jest.fn().mockReturnValue(1000),
      isSmallDevice: false
    },
    deviceInfo: {
      getDeviceId: createUUID,
      getUniqueId: createUUID,
      getVersion: () => '0.0',
      getBuildNumber: () => '0.1',
      getDeviceModel: () => 'iPhone 11',
      getLocales: { languageTag: 'en-US' },
      getBaseOs: async () => 'iOS',
      getSystemVersion: () => '11.0',
      getDeviceToken: () => createUUID(),
      getDeviceTokenAsync: async () => createUUID(),
      getDeviceOsType: async () => 'iOS',
      getAppVersion: () => '0.0.0.1',
      getDeviceLanguage: () => 'en-US',
      getDeviceOsVersion: () => '11.0',
      getFCMTokenAsync: async () => createUUID(),
      getTimeZone: () => 'testing-timezone',
      getReadableVersion: () => '1.1.1.1',
      onFCMTokenRefresh: jest.fn()
    },
    netInfo: {
      getState: jest.fn().mockResolvedValue({ isConnected: true }),
      subscribeToConnectionChanges: jest.fn(),
      waitForConnection: jest.fn().mockResolvedValue(undefined)
    },
    linking: {
      openURL: jest.fn(() => Promise.resolve()),
      canOpenURL: jest.fn(),
      checkURLScheme: jest.fn().mockResolvedValue(true),
      getInitialURL: jest.fn(),
      addEventListener: jest.fn()
    },
    buttonSdk: {
      configure: jest.fn(),
      setUserId: jest.fn(),
      clearData: jest.fn(),
      purchaseRequest: jest.fn()
    },
    platform: {
      os: 'default',
      select: jest.fn(config => config?.default)
    },
    fireBaseMessaging: {
      getToken: jest.fn().mockReturnValue(createUUID())
    },
    hash: {
      buildSHA256: jest.fn().mockResolvedValue('hash')
    },
    restart: jest.fn(),
    cookieManager: {
      clearAll: jest.fn()
    },
    base64: {
      encode: jest.fn((...args) => `b64 encoded: ${JSON.stringify(args)}`),
      decode: jest.fn((...args) => `b64 decoded: ${JSON.stringify(args)}`)
    },
    getRuntimeEnv: jest.fn().mockResolvedValue(RUNTIME_ENV.DEV_TEST),
    messaging: {
      hasPermission: jest.fn().mockReturnValue(true),
      requestPermission: jest.fn()
    },
    pushNotificationsAuthStatus: {
      NOT_DETERMINED: -1,
      DENIED: 0,
      AUTHORIZED: 1,
      PROVISIONAL: 2
    },
    sailthru: {
      getMessages: jest
        .fn()
        .mockResolvedValue([getNewOnMaxMessage_1(), getNewOnMaxMessage_4(), getNewOnMaxMessage_2(), getNewOnMaxMessage_3(), getSailthruMessage_1()]),
      markMessageAsRead: jest.fn().mockResolvedValue(true),
      getInitialNotification: jest.fn(),
      onNotificationTapped: jest.fn(cb => {
        cb({});
        return () => {};
      }) as any
    },
    reactNativePermission: require('react-native-permissions/mock'),
    geolocation: {
      addListener: jest.fn(),
      getCurrentPosition: jest.fn(),
      removeListeners: jest.fn(),
      requestAuthorization: jest.fn(),
      setConfiguration: jest.fn(),
      startObserving: jest.fn(),
      stopObserving: jest.fn()
    }
  } as any,
  stateSnapshot: {
    set: jest.fn(),
    get: jest.fn().mockReturnValue(initialStateGetter())
  },
  logger: {
    log: jest.fn(),
    info: jest.fn(),
    debug: jest.fn(),
    warn: jest.fn(),
    error: jest.fn(),
    success: jest.fn(),
    color: jest.fn(),
    assert: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    format: jest.fn()
  } as any,
  eventTrackerService: {
    trackDataUser: jest.fn(),
    isTrackingEnabled: jest.fn().mockResolvedValue(true),
    rZero: {
      trackView: jest.fn(),
      trackEvent: jest.fn(),
      flush: jest.fn(),
      setUser: jest.fn()
    },
    forterSDK: {
      setDevLogsEnabled: jest.fn(),
      getDeviceUniqueID: jest.fn(cb => cb()),
      init: jest.fn((a, b, c) => c()),
      trackNavigation: jest.fn(),
      trackNavigationWithExtraData: jest.fn(),
      trackAction: jest.fn(),
      trackActionWithJSON: jest.fn(),
      fraudTrackNavigationEvent: jest.fn(),
      fraudTrackAction: jest.fn()
    },
    tealiumSDK: {
      track: jest.fn(),
      trackView: jest.fn()
    },
    appsFlyerSDK: {
      updateServerUninstallToken: jest.fn(),
      setCustomerUserId: jest.fn().mockImplementation((_, cb) => cb('success')),
      addDeepLinkEventListener: jest.fn(),
      removeDeepLinkEventListener: jest.fn(),
      clearDeepLinkEventListener: jest.fn(),
      trackConversionEvent: jest.fn()
    }
  } as any,
  ccpaService: {
    showConsentBanner: jest.fn().mockResolvedValue(null),
    showPreferenceCenter: jest.fn()
  } as any,
  remoteConfigService: {
    getValue: jest.fn(async key => ENV.REMOTE_CONFIG.DEFAULT_CONFIG[key]),
    fetchAndActivate: jest.fn().mockResolvedValue(null)
  } as any,
  authService: {
    auth: {
      refreshToken: jest.fn().mockResolvedValue(true)
    },
    webAuth: {
      authorize: jest.fn(),
      clearSession: jest.fn()
    }
  } as any
});
