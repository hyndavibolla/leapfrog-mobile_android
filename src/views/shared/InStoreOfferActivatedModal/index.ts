export { default as InStoreOfferActivatedModal, InStoreOfferActivatedModalKey } from './InStoreOfferActivatedModal';
