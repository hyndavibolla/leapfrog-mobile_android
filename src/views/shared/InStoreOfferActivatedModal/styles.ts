import { css } from '@emotion/native';

import { COLOR, FONT_FAMILY, FONT_SIZE } from '../../../constants';

export const styles = {
  container: css`
    align-items: center;
  `,
  header: css`
    flex-direction: row;
    justify-content: center;
    align-items: center;
    margin-bottom: 26px;
  `,
  title: css`
    text-align: center;
    color: ${COLOR.BLACK};
  `,
  body: css`
    margin-top: 10px;
  `,
  bodyText: css`
    text-align: center;
    font-size: ${FONT_SIZE.SMALLER};
    color: ${COLOR.DARK_GRAY};
  `,
  list: css`
    margin-bottom: 20px;
  `,
  listItem: css`
    flex-direction: row;
    justify-content: center;
    align-items: center;
  `,
  listItemIcon: css`
    margin-right: 5px;
  `,
  plus: css`
    color: ${COLOR.BLACK};
    font-size: ${FONT_SIZE.MEDIUM};
    line-height: ${FONT_SIZE.HUGE};
    font-family: ${FONT_FAMILY.HEAVY};
    margin: 0 14px;
  `,
  linkBtnInnerContainer: css`
    padding: 16px 120px;
    margin-top: 40px;
  `,
  linkBtnText: css`
    font-family: ${FONT_FAMILY.MEDIUM};
    font-size: ${FONT_SIZE.SMALLER};
  `
};
