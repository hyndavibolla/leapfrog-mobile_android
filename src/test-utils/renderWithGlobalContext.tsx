import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
