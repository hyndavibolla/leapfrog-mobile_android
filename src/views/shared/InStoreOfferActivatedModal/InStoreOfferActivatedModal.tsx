import React, { memo, useCallback } from 'react';
import { View, Text } from 'react-native';

import { useTestingHelper } from '_utils/useTestingHelper';
import { DateLike } from '_models/general';
import { getDateDiffInDays } from '_utils/getDateDiffInDays';
import RestaurantFallback from '_assets/in-store-offer/restaurantFallback.svg';
import { Icon } from '_commons/components/atoms/Icon';
import { ICON, FONT_SIZE, COLOR } from '_constants';
import { Button } from '_components/Button';
import { BrandLogo } from '_components/BrandLogo';
import { Title, TitleType } from '_components/Title';

import { styles } from './styles';

export interface Props {
  brandLogo?: string;
  brandName: string;
  activeUntil: DateLike;
  onActionPress: () => void;
  street: string;
}

export const InStoreOfferActivatedModalKey = 'in-store-offer-activated-modal';

const InStoreOfferActivatedModal = ({ brandLogo, brandName, activeUntil, onActionPress, street }: Props) => {
  const { getTestIdProps } = useTestingHelper(InStoreOfferActivatedModalKey);
  const diffDays = getDateDiffInDays(Date.now(), activeUntil);

  type ItemViewProps = { value: string; check?: boolean };
  const ItemView = useCallback(
    ({ value, check }: ItemViewProps) => (
      <View style={styles.listItem}>
        {check && <Icon name={ICON.TICK} color={COLOR.PURPLE} size={FONT_SIZE.PETITE} style={styles.listItemIcon} />}
        <Text style={styles.bodyText}>{value}</Text>
      </View>
    ),
    []
  );

  return (
    <View style={styles.container} {...getTestIdProps('container')}>
      <View style={styles.header}>
        <BrandLogo image={brandLogo} Fallback={RestaurantFallback} size={60} />
        <Text style={styles.plus}>+</Text>
        <Icon name={ICON.OFFER_CIRCLE} color={COLOR.PURPLE} size={FONT_SIZE.XL} />
      </View>
      <Title style={styles.title} type={TitleType.HEADER}>
        Your {brandName} offer is activated!
      </Title>
      <View style={styles.body}>
        <Text style={styles.bodyText}>What do I do next?</Text>
        <View style={styles.list}>
          <ItemView value="Go to the restaurant" check />
          <ItemView value="Purchase food" check />
          <ItemView value="Enjoy! ðŸŽ‰" />
        </View>
        {!isNaN(diffDays) && diffDays > 0 && (
          <Text style={styles.bodyText} {...getTestIdProps('text-diff-day')}>
            This offer is available for {diffDays} days{street ? ` for the restaurant at ${street}.` : '.'}
          </Text>
        )}
      </View>

      <Button innerContainerStyle={styles.linkBtnInnerContainer} textStyle={styles.linkBtnText} onPress={onActionPress} {...getTestIdProps('action-btn')}>
        Got it!
      </Button>
    </View>
  );
};

export default memo(InStoreOfferActivatedModal);
