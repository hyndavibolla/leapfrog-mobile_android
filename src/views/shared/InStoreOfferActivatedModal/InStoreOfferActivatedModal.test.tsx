import React from 'react';
import { fireEvent } from '@testing-library/react-native';
import { GeneralModel } from '../../../models';
import { getLocalOffers_1 } from '../../../test-utils/entities';
import { getMockDeps } from '../../../test-utils/getMockDeps';
import { renderWithGlobalContext } from '../../../test-utils/renderWithGlobalContext';

import InStoreOfferActivatedModal, { Props, InStoreOfferActivatedModalKey } from './InStoreOfferActivatedModal';

describe('InStoreOfferActivatedModal', () => {
  let deps: GeneralModel.Deps;
  let props: Props;
  let dateNowSpy;

  beforeEach(() => {
    deps = getMockDeps();
    const { offers } = getLocalOffers_1();
    const [
      {
        brandName,
        brandLogo,
        activeUntil,
        merchant: {
          address: { street }
        }
      }
    ] = offers;
    props = {
      brandName,
      brandLogo,
      activeUntil,
      street,
      onActionPress: jest.fn()
    };
    dateNowSpy = jest.spyOn(Date, 'now').mockImplementation(() => 1629903600000);
  });

  afterAll(() => {
    dateNowSpy.mockRestore();
  });

  it('should render', () => {
    const { toJSON } = renderWithGlobalContext(<InStoreOfferActivatedModal {...props} />, deps);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render without brand logo and address', () => {
    props.brandLogo = null;
    props.street = null;
    const { toJSON } = renderWithGlobalContext(<InStoreOfferActivatedModal {...props} />, deps);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should have a pressable action button', () => {
    const { getByTestId } = renderWithGlobalContext(<InStoreOfferActivatedModal {...props} />, deps);
    fireEvent.press(getByTestId(`${InStoreOfferActivatedModalKey}-action-btn`));
    expect(props.onActionPress).toBeCalled();
  });

  it('should show text of available days', () => {
    const { getByTestId } = renderWithGlobalContext(<InStoreOfferActivatedModal {...props} />, deps);
    expect(getByTestId(`${InStoreOfferActivatedModalKey}-text-diff-day`)).toBeTruthy();
  });

  it('should hide text of available days', () => {
    props.activeUntil = null;
    const { queryByTestId } = renderWithGlobalContext(<InStoreOfferActivatedModal {...props} />, deps);
    expect(queryByTestId(`${InStoreOfferActivatedModalKey}-text-diff-day`)).toBeNull();
  });
});
