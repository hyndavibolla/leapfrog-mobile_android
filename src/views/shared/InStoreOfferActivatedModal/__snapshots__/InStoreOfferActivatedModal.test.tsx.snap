// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InStoreOfferActivatedModal should render 1`] = `
<View
  accessibilityLabel="in-store-offer-activated-modal-container"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
    }
  }
  testID="in-store-offer-activated-modal-container"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 26,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 60,
            "width": 60,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "alignItems": "center",
              "borderBottomLeftRadius": 60,
              "borderBottomRightRadius": 60,
              "borderTopLeftRadius": 60,
              "borderTopRightRadius": 60,
              "height": "100%",
              "justifyContent": "center",
              "width": "100%",
            },
          ]
        }
      >
        <FastImageView
          accessibilityLabel="brand-logo-image"
          accessible={true}
          onFastImageError={[Function]}
          onFastImageLoad={[Function]}
          onFastImageLoadEnd={[Function]}
          onFastImageLoadStart={[Function]}
          resizeMode="cover"
          source={
            Object {
              "priority": "high",
              "uri": "http://www.local-offer.com/logo/1",
            }
          }
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          testID="brand-logo-image"
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Heavy",
          "fontSize": 22,
          "lineHeight": 26,
          "marginBottom": 0,
          "marginLeft": 14,
          "marginRight": 14,
          "marginTop": 0,
        }
      }
    >
      +
    </Text>
    <Text
      accessibilityLabel="icon-text"
      accessible={true}
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#B361DF",
            "fontSize": 60,
          },
          undefined,
          Object {
            "fontFamily": "sywmax",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="icon-text"
    >
      î¥„
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 20,
          "lineHeight": 30,
        },
        Object {
          "color": "#262C2F",
          "textAlign": "center",
        },
      ]
    }
    type="titleHeader"
  >
    Your 
    local-offer-brand-1
     offer is activated!
  </Text>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#909BA0",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      What do I do next?
    </Text>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessibilityLabel="icon-text"
          accessible={true}
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#B361DF",
                "fontSize": 14,
              },
              Object {
                "marginRight": 5,
              },
              Object {
                "fontFamily": "sywmax",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="icon-text"
        >
          î¤¸
        </Text>
        <Text
          style={
            Object {
              "color": "#909BA0",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Go to the restaurant
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessibilityLabel="icon-text"
          accessible={true}
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#B361DF",
                "fontSize": 14,
              },
              Object {
                "marginRight": 5,
              },
              Object {
                "fontFamily": "sywmax",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="icon-text"
        >
          î¤¸
        </Text>
        <Text
          style={
            Object {
              "color": "#909BA0",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Purchase food
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#909BA0",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Enjoy! ðŸŽ‰
        </Text>
      </View>
    </View>
    <Text
      accessibilityLabel="in-store-offer-activated-modal-text-diff-day"
      accessible={true}
      style={
        Object {
          "color": "#909BA0",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
      testID="in-store-offer-activated-modal-text-diff-day"
    >
      This offer is available for 
      108
       days
       for the restaurant at 2307 Beverley Rd, Brooklyn, NY 11226, United States.
    </Text>
  </View>
  <View
    accessibilityLabel="in-store-offer-activated-modal-action-btn"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="in-store-offer-activated-modal-action-btn"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#0066CC",
            "borderBottomLeftRadius": 30,
            "borderBottomRightRadius": 30,
            "borderTopLeftRadius": 30,
            "borderTopRightRadius": 30,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          },
          Object {
            "marginTop": 40,
            "paddingBottom": 16,
            "paddingLeft": 120,
            "paddingRight": 120,
            "paddingTop": 16,
          },
        ]
      }
    >
      <View
        style={false}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "SFProDisplay-Medium",
                  "fontSize": 16,
                },
                undefined,
              ],
            ]
          }
        >
          Got it!
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`InStoreOfferActivatedModal should render without brand logo and address 1`] = `
<View
  accessibilityLabel="in-store-offer-activated-modal-container"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
    }
  }
  testID="in-store-offer-activated-modal-container"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 26,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 60,
            "width": 60,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        RestaurantFallback
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Heavy",
          "fontSize": 22,
          "lineHeight": 26,
          "marginBottom": 0,
          "marginLeft": 14,
          "marginRight": 14,
          "marginTop": 0,
        }
      }
    >
      +
    </Text>
    <Text
      accessibilityLabel="icon-text"
      accessible={true}
      allowFontScaling={false}
      selectable={false}
      style={
        Array [
          Object {
            "color": "#B361DF",
            "fontSize": 60,
          },
          undefined,
          Object {
            "fontFamily": "sywmax",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
      testID="icon-text"
    >
      î¥„
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#262C2F",
          "fontFamily": "SFProDisplay-Bold",
          "fontSize": 20,
          "lineHeight": 30,
        },
        Object {
          "color": "#262C2F",
          "textAlign": "center",
        },
      ]
    }
    type="titleHeader"
  >
    Your 
    local-offer-brand-1
     offer is activated!
  </Text>
  <View
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#909BA0",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
    >
      What do I do next?
    </Text>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessibilityLabel="icon-text"
          accessible={true}
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#B361DF",
                "fontSize": 14,
              },
              Object {
                "marginRight": 5,
              },
              Object {
                "fontFamily": "sywmax",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="icon-text"
        >
          î¤¸
        </Text>
        <Text
          style={
            Object {
              "color": "#909BA0",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Go to the restaurant
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          accessibilityLabel="icon-text"
          accessible={true}
          allowFontScaling={false}
          selectable={false}
          style={
            Array [
              Object {
                "color": "#B361DF",
                "fontSize": 14,
              },
              Object {
                "marginRight": 5,
              },
              Object {
                "fontFamily": "sywmax",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="icon-text"
        >
          î¤¸
        </Text>
        <Text
          style={
            Object {
              "color": "#909BA0",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Purchase food
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#909BA0",
              "fontSize": 16,
              "textAlign": "center",
            }
          }
        >
          Enjoy! ðŸŽ‰
        </Text>
      </View>
    </View>
    <Text
      accessibilityLabel="in-store-offer-activated-modal-text-diff-day"
      accessible={true}
      style={
        Object {
          "color": "#909BA0",
          "fontSize": 16,
          "textAlign": "center",
        }
      }
      testID="in-store-offer-activated-modal-text-diff-day"
    >
      This offer is available for 
      108
       days
      .
    </Text>
  </View>
  <View
    accessibilityLabel="in-store-offer-activated-modal-action-btn"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="in-store-offer-activated-modal-action-btn"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#0066CC",
            "borderBottomLeftRadius": 30,
            "borderBottomRightRadius": 30,
            "borderTopLeftRadius": 30,
            "borderTopRightRadius": 30,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 8,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 8,
          },
          Object {
            "marginTop": 40,
            "paddingBottom": 16,
            "paddingLeft": 120,
            "paddingRight": 120,
            "paddingTop": 16,
          },
        ]
      }
    >
      <View
        style={false}
      >
        <Text
          style={
            Array [
              Object {
                "color": "#909BA0",
                "fontFamily": "SFProDisplay-Medium",
                "fontSize": 16,
                "lineHeight": 22,
              },
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 14,
                  "textAlign": "center",
                },
                Object {
                  "fontFamily": "SFProDisplay-Medium",
                  "fontSize": 16,
                },
                undefined,
              ],
            ]
          }
        >
          Got it!
        </Text>
      </View>
    </View>
  </View>
</View>
`;
